File: ECnotes/Sem2/LivExamples/CS/Ch19CartProfile/App_Code/CartItem.cs
using System;

// You must mark this class as serializable
// to get the Profile feature to work
[Serializable]
public class CartItem
{
    public Product Product;
    public int Quantity;

    public String display
    {
        get
        {
            return this.Product.Name + " (" + this.Quantity.ToString()
                + " at " + this.Product.UnitPrice.ToString("c") + " each)";
        }
    }
}

File: ECnotes/Sem2/LivExamples/VB/Ch19CartProfile/App_Code/CartItem.vb
Imports Microsoft.VisualBasic

<Serializable()> _
Public Class CartItem
    Public Product As Product
    Public Quantity As Integer

    Public Function Display() As String
        Return Product.Name + " (" + Quantity.ToString() _
            + " at " + FormatCurrency(Product.UnitPrice) + " each)"

    End Function
End Class
